src/Illuminate/Http/Resources/ConditionallyLoadsAttributes.php

```diff
@@ -223,4 +226,32 @@ protected function transform($value, callable $callback, $default = null)
             $value, $callback, func_num_args() === 3 ? $default : new MissingValue
         );
     }
+
+    /**
+     * Verify if the resource keys should be preserved.
+     *
+     * @return bool
+     */
+    protected function shouldPreserveKeys()
+    {
+        if (property_exists($this, 'preserveKeys') && $this->preserveKeys === true) {
+            return true;
+        }
+
+        if ($this instanceof ResourceCollection) {
+            try {
+                $collectedResource = new ReflectionClass($this->collects);
+            } catch (Exception $exception) {
```

Comment:
should probably catch a reflection exception here?

Reply:
Changed the approach and moved this method to another file. But made the change you suggested. Thanks!